app/src/main/java/com/realityexpander/tasky/agenda_feature/presentation/agenda_screen/AgendaScreen.kt

```diff
@@ -540,222 +557,218 @@ fun AgendaScreenContent(
                 .tinyHeight()
         )
 
+        // • EMPTY AGENDA INDICATOR
+        val showEmptyIndicator = remember { mutableStateOf(false) }
+        LaunchedEffect(agendaItems.isEmpty()) {
+            showEmptyIndicator.value = false
+
+            if (agendaItems.isEmpty()) {
+                delay(400) // to prevent flickering when agenda items are loading
+                showEmptyIndicator.value = true
+            }
+        }
+
         // • SHOW AGENDA ITEMS LIST
-        if (agendaItems.isNotEmpty()) {
-            LazyColumn(
-                state = scrollState,
-                modifier = Modifier
-                    .background(color = MaterialTheme.colors.surface)
-                    .fillMaxSize()
-                    .padding(start = DP.tiny, end = DP.tiny)
+        LazyColumn(
+            state = scrollState,
+            modifier = Modifier
+                .background(color = MaterialTheme.colors.surface)
+                .fillMaxSize()
+                .padding(start = DP.tiny, end = DP.tiny)
 
-            ) {
-                val sortedAgendaItems = agendaItems.sortedBy { agendaItem ->
-                    agendaItem.startTime
-                }
-                val agendaItemsBeforeNow = sortedAgendaItems.filter { agendaItem ->
-                    agendaItem.startTime.isBefore(zonedDateTimeNow)
-                }
-                val agendaItemsAfterNow = sortedAgendaItems.filter { agendaItem ->
-                    agendaItem.startTime.isAfter(zonedDateTimeNow)
-                }
+        ) {
+            val sortedAgendaItems = agendaItems.sortedBy { agendaItem ->
```

Comment:
Sorting inside a LazyColumn means you'll be sorting on every recomposition

Its better to either;
A- Sort it on viewmodel
B- Add a remember to the sort. Since its based on a state item, you probably need derivedStateOf